Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,460 +22,653 @@
-							std::cout << "Warning\t| IfcSurfaceOfLinearExtrusion not implemented." << std::endl;
-#endif
-							return;
-						}
+					for(unsigned int i = 0; i < vInnerBoundaries.size(); ++i) {
+						EXPRESSReference<typename IfcEntityTypesT::IfcCurve>& innerCurve = vInnerBoundaries[i];
+					}
 
-						// (2/2) IfcSurfaceOfRevolution SUBTYPE of IfcSweptSurface
-						std::shared_ptr<typename IfcEntityTypesT::IfcSurfaceOfRevolution> surface_of_revolution =
-							std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcSurfaceOfRevolution>(swept_surface);
+					*/
 
-						if(surface_of_revolution) {
-							// Get axis position.
-							std::shared_ptr<typename IfcEntityTypesT::IfcAxis1Placement>& axis_position =
-								surface_of_revolution->AxisPosition.lock(); // TODO: check member name erledigt
+					throw oip::UnhandledException(surface);
+				}
 
-	  // TODO: implement		// Interpret values and calculate.
-#ifdef _DEBUG
-							std::cout << "Warning\t| IfcSurfaceOfRevolution not implemented." << std::endl;
-#endif
-							return;
+				/*! \brief  Converts \c IfcCurveBoundedSurface to ...
+				\param	surface	\c IfcCurveBoundedSurface entity to be interpreted.
+				\param	pos
+				\param	polylineData
+				\note	The \c IfcCurveBoundedSurface is a subtype of \c IfcBoundedSurface.
+				*/
+
+				void convertIfcCurveBoundedSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcCurveBoundedSurface>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) const throw(...) {
+
+					/* TO DO: Finish implementation of convertIfcCurveBoundedSurface.
+
+					// Get basis surface, boundaries and implicit outer.
+					EXPRESSReference<typename IfcEntityTypesT::IfcSurface> basisSurface = surface->BasisSurface;
+					EXPRESSReference<typename IfcEntityTypesT::IfcBoundaryCurve> boundaries = surface->Boundaries;
+					std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcBoundaryCurve>> vBoundaries;
+
+					vBoundaries.resize(surface->Boundaries.size());
+					std::transform(
+						surface->Boundaries.begin(),
+						surface->Boundaries.end(),
+						vBoundaries.begin(),
+						[](auto& it)
+						{ return it.lock(); });
+					EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint> controlPointList = surface->ControlPointsList;
+
+					bool implicitOuter = surface->ImplicitOuter;
+
+					*/
+
+					throw oip::UnhandledException(surface);
+				}
+
+				/*! \brief  Converts \c IfcRectangularTrimmedSurface to ...
+				\param	surface	\c IfcRectangularTrimmedSurface entity to be interpreted.
+				\param	pos
+				\param	polylineData
+				\note	The \c IfcRectangularTrimmedSurface is a subtype of \c IfcBoundedSurface.
+				*/
+
+				void convertIfcRectangularTrimmedSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcRectangularTrimmedSurface>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) const throw(...) {

```

Comment:
Again, pos and maybe also polyline data should be returned from the function.

See here how to return multiple elements:
https://github.com/tumcms/Open-Infra-Platform/blob/75a2c0b4fda902bd97785cacb2f7498ad235fd6c/Core/src/IfcGeometryConverter/PlacementConverter.h#L549-L554

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/1ee08611e218ac8b03e6fabd0e214cad06a096b1. On request by @pjanck only polylineData as return parameter.